var/spack/repos/builtin/packages/root/package.py

```diff
@@ -507,23 +506,27 @@ def cmake_args(self):
                            spec['python'].command.path)
         return options
 
-    def setup_environment(self, spack_env, run_env):
-        run_env.set('ROOTSYS', self.prefix)
-        run_env.set('ROOT_VERSION', 'v{0}'.format(self.version.up_to(1)))
-        run_env.prepend_path('PYTHONPATH', self.prefix.lib)
+    def setup_build_environment(self, env):
         if 'lz4' in self.spec:
-            spack_env.append_path('CMAKE_PREFIX_PATH',
-                                  self.spec['lz4'].prefix)
-        spack_env.set('SPACK_INCLUDE_DIRS', '', force=True)
-
-    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
-        spack_env.set('ROOTSYS', self.prefix)
-        spack_env.set('ROOT_VERSION', 'v{0}'.format(self.version.up_to(1)))
-        spack_env.prepend_path('PYTHONPATH', self.prefix.lib)
-        spack_env.prepend_path('PATH', self.prefix.bin)
-        spack_env.append_path('CMAKE_MODULE_PATH', '{0}/cmake'
-                              .format(self.prefix))
-        run_env.set('ROOTSYS', self.prefix)
-        run_env.set('ROOT_VERSION', 'v{0}'.format(self.version.up_to(1)))
-        run_env.prepend_path('PYTHONPATH', self.prefix.lib)
-        run_env.prepend_path('PATH', self.prefix.bin)
+            env.append_path('CMAKE_PREFIX_PATH',
```

Comment:
interesting, this should be automatically be set by Spack, no? cc @alalazo

Reply:
There's an issue if `lz4` is found from `packages.yaml`, I think. It was definitely found to be necessary, and similar things have been done in other recipes.